source/flags.cpp

```diff
@@ -1,23 +1,20 @@
 #include "flags.h"
+#include "libtww/d/com/d_com_inf_game.h"
 
 GZFlag g_flags[FLAG_AMNT] = {

```

Comment:
is there any reason to store these flags permanently? you should be able to check the state of the flag on menu init so that the toggle always represents the actual flag state instead of using a intermediate flag

Reply:
I think doing it this way makes it easier for others to get involved and add flags they want, which is more important imo.  Unless I'm misunderstanding what you're suggesting

Reply:
my suggestion is just to remove the need to store unnecessary data like the flag key globally when its only used by the 1 menu, since as more flags are added its going to unnecessarily eat up more of the valuable memory.  it may make it a tad more tedious to implement, but the benefits should probably outweigh the negatives

you could just have an array of standard bools for the toggle vars, then update the state the same way you're doing but without keeping track of the redundant `active` var and manually writing the flag key. 
like
```c++
void updateFlags() {
    g_flags[0] = dComIfGs_isEventBit(0x3101);
    // and so on...
}
```
that way the minimal amount of info is stored. this obv sucks if you're doing a massive amount of flags, but i'd argue that none of the menus should be dealing with massive amounts of flags to begin with. if they're split into appropriate categories and prioritized by importance then a massive amount shouldnt be needed in the first place

Reply:
Lots of good points.  I'll make your suggested changes.

Reply:
Just pushed your suggestion.  Thanks for the constructive feedback!